### DISCUSSION TITLE
Using middleware in single route

### DISCUSSION THREAD

**Discussion by tombracho** [go-chi/chi#810] [General]
Tell me please, why the Chain().HandlerFunc() returns type 'Handler' instead of 'HandlerFunc'?
I use it to add some middleware for single route, and i need to call ServeHTTP method. Like this:

r := chi.NewRouter()
r.Get("/", chi.Chain(middleware.Logger).HandlerFunc(homeHandler).ServeHTTP)

**Comment by VojtechVitek** [comment]
`r.With(mw).Get("/", handler)`

**Reply by tombracho** [reply]
Thanks a lot

**Comment by tombracho** [comment]
Yeah, ingenious is simple
